contracts/StablecoinConverter.sol

```diff
@@ -260,31 +279,39 @@ contract StablecoinConverter is EpochTokenLocker {
                     currentPrices[order.buyToken],
                     currentPrices[order.sellToken]
                 );
+                buyVolume = uint128(buyVolume.mul(feeDenominator - 1) / feeDenominator);
                 revertRemainingOrder(owner, orderId, sellVolume);
                 subtractBalance(owner, tokenIdToAddressMap(order.buyToken), buyVolume);
             }
         }
     }
 
     function findPriceIndex(uint16 index, uint16[] memory tokenIdForPrice) private pure returns (uint) {
+        // return fee token
+        if (index == 0) {
+            return 0;
+        }
+        // binary search for the other tokens
         uint leftValue = 0;
         uint rightValue = tokenIdForPrice.length - 1;
         while (rightValue >= leftValue) {
             uint middleValue = leftValue + (rightValue-leftValue) / 2;
             if (tokenIdForPrice[middleValue] == index) {
-                return middleValue;
+                return middleValue + 1;
             } else if (tokenIdForPrice[middleValue] < index) {
                 leftValue = middleValue + 1;
             } else {
+                require(middleValue > 0, "Price not provided for token, underflow would have happened");
```

Comment:
In what scenario does this happen?

Reply:
Yeah, it's not a usual case. Check out the unit test for it.